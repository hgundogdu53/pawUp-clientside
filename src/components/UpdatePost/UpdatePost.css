.update-post-section {
    transition: 0.3s;
    border-radius: 5px;
    width: 22.5rem;
    margin-top: 0.5rem;
    background-color: rgba(202,244,241, 0.2);
    width: 30rem;
}

.update-post-section:hover {
    box-shadow: 0 8px 16px 0 rgba(255, 255, 255, 0.151);
}


.sub-btn,
.res-btn {
    background-color: #fff;
    font-weight: bold;
    text-decoration: none;
    margin-top: 15px;
    padding: 7.5px 20px;
    text-transform: uppercase;
    text-decoration: none;
    display: inline-block;
    transition: all .2s;
    border-radius: 100px;
}

.sub-btn:hover, .res-btn:hover {
    cursor: pointer;
    transform: translateY(-2.5px);
}

.sub-btn:active, .res-btn:hover {
    transform: translateY(-1px);
}

.update-post-form {
    padding: 1rem;
}

.update-post-form label, .update-post-form input {
    display: block;
    width: 100%;
    text-align: start;
    margin-top: 0.5rem;
}

.update-post-form input {
    border: 1px solid #ccc;
    padding: 0.2rem 0.5rem;
    font: inherit;
}

.update-post-form input:focus {
    outline: none;
    border-color: rgb(4, 127, 168);
}

.update-post-form label {
    font-weight: bold;
}

.update-post-section {
    padding: 5px 75px;
}

.update-post-title {
    text-align: center;
    font-size: 1.75rem;
}

.update-post-form button[type="submit"],
.update-post-form button[type="reset"] {
    display: inline-block;
    margin-top: 1rem;

}


@media only screen and (min-width: 40rem) {
    .signup_form {
        margin: auto;
        width: 35rem;
    }
}

@media only screen and (min-width: 782px) {
    .update-post-section {
        width: 40rem;
    }
}