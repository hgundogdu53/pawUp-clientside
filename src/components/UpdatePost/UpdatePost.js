import React from 'react';
import './UpdatePost.css';

function UpdatePost(props) {
    const post = props.posts.filter(post => post.id === props.postId)
    if (post) {
        return (
            <main className='flex-container'>
                <header>
                    <h1 className='update-post-title'>Update Post</h1>
                </header>
                <section className='update-post-section'>
                    <form id="update-post-form" className='update-post-form' onSubmit={props.updatePost}>
                        <div className="form-section">
                            <label htmlFor="pet_name">Pet Name</label>
                            <input type="text" name="pet_name" defaultValue={post[0].pet_name} required />
                        </div>
                        <div className="form-section">
                            <label htmlFor="owner_email">Email</label>
                            <input type="email" name="owner_email" defaultValue={post[0].email} required />
                        </div>
                        <div className="form-section">
                            <label htmlFor="breed">Type of Pet</label>
                            <input type="text" name="type_of_pet" defaultValue={post[0].type_of_pet} required />
                        </div>
                        <div className="form-section">
                            <label htmlFor="lifestyle">Hobbies</label>
                            <textarea name="lifestyle" rows="15" placeholder="Please enter a description of your dogs needs and lifestyle" defaultValue={post[0].hobbies} required></textarea>
                        </div>
                        <button type="submit" className='sub-btn'>Submit</button>
                        <button type="reset" className='res-btn'>Reset</button>
                    </form>
                </section>
            </main>
        )
    } else {
        return (
            <main>
                <h1>No Post To Update!</h1>
                <h2>Try Again</h2>
            </main>
        )
    }

}

export default UpdatePost;