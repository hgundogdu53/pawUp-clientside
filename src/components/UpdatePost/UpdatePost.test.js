import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UpdatePost from './UpdatePost';

describe(`UpdatePost component`, () => {
    it('renders without crashing', () => {
        const post = [
            {
                pet_name: 'muffin',
                email: 'hgundogdu53@gmail.com',
                type_of_pet: 'cat',
                birthdate: '2018-06-18',
                hobbies: 'Small dog that loves to run around. Will fit well in smaller homes as well as larger homes. Will need atleast an hour of exercise to keep out of trouble.',
                owner_id: 1
            },

        ]
        const wrapper = shallow(<UpdatePost posts={post} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});