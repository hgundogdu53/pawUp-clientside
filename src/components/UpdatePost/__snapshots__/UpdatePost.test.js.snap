// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePost component renders without crashing 1`] = `
<main
  className="flex-container"
>
  <header>
    <h1
      className="update-post-title"
    >
      Update Post
    </h1>
  </header>
  <section
    className="update-post-section"
  >
    <form
      className="update-post-form"
      id="update-post-form"
    >
      <div
        className="form-section"
      >
        <label
          htmlFor="pet_name"
        >
          Pet Name
        </label>
        <input
          defaultValue="muffin"
          name="pet_name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="owner_email"
        >
          Email
        </label>
        <input
          defaultValue="hgundogdu53@gmail.com"
          name="owner_email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="type_of_pet"
        >
          Type of Pet
        </label>
        <input
          defaultValue="cat"
          name="type_of_pet"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="lifestyle"
        >
          Hobbies
        </label>
        <textarea
          defaultValue="Small dog that loves to run around. Will fit well in smaller homes as well as larger homes. Will need atleast an hour of exercise to keep out of trouble."
          name="lifestyle"
          placeholder="Please enter a description of your dogs needs and lifestyle"
          required={true}
          rows="15"
        />
      </div>
      <button
        className="sub-btn"
        type="submit"
      >
        Submit
      </button>
      <button
        className="res-btn"
        type="reset"
      >
        Reset
      </button>
    </form>
  </section>
</main>
`;
